### PR TITLE
Create Gaming COG - !bgg and !scry commands

### PR DESCRIPTION
**Pull request by bradleysigma**
image

image

### CODE REVIEW

File: uqcsbot/gaming.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+            description = description[:max_message_length] + u"\u2026"
+        embed.add_field(name="Description", inline=False, value=description)
+        embed.add_field(name="Board Game Geek Link", inline=False,
+                        value=f"https://boardgamegeek.com/boardgame/{parameters.get('identity'):s}")
+        embed.set_thumbnail(url=parameters.get('image'))
+        return embed
+
+    @commands.command()
+    @loading_status
+    # TODO: @loading_status
```
Comment (JamesDearlove) [inline uqcsbot/gaming.py:181]: Is there a reason that a todo for `@loading_status` exists?

File: uqcsbot/gaming.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    async def bgg(self, ctx: commands.Context, *board_game: str):
+        """
+        `!bgg board_game` - Gets the details of `board_game` from Board Game Geek
+        """
```
Comment (JamesDearlove) [inline uqcsbot/gaming.py:185]: The help command already injects how to use the command, we only require the description of what the command does.

File: uqcsbot/gaming.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        if parameters is None:
+            await ctx.send("Something has gone wrong.")
+            return
+
+        embed = self.format_board_game_parameters(parameters)
+        await ctx.send(embed=embed)
+
+    @commands.command()
+    @loading_status
+    # TODO: @loading_status
```
Comment (JamesDearlove) [inline uqcsbot/gaming.py:202]: As above regarding todo?

File: uqcsbot/gaming.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    async def scry(self, ctx: commands.Context, *card: str):
+        """
+        `!scry [card]` - Returns the Magic: the Gathering card that matches (partially
+        or fully) the given argument (or a random card if no argument given)
+        """
```
Comment (JamesDearlove) [inline uqcsbot/gaming.py:207]: As above regarding help text.

File: uqcsbot/gaming.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
+from discord.ext import commands
+import discord
+import logging
+from uqcsbot.bot import UQCSBot
+from typing import List, Tuple, Optional
```
Comment (JamesDearlove) [inline uqcsbot/gaming.py:5]: Unneeded imports of logging, List, Tuple

**Review by JamesDearlove — CHANGES_REQUESTED** [review]
Seems mostly good to me, mostly cosmetic changes.

However, for whatever reason searching for coup throws errors. Quite ironic if you ask me.

File: uqcsbot/gaming.py (hunk 1 above)
Comment (bradleysigma) [reply uqcsbot/gaming.py:181]: Because I copied it from https://github.com/UQComputingSociety/uqcsbot-discord/blob/main/uqcsbot/events.py#L282

Comment (bradleysigma) [reply uqcsbot/gaming.py]: Fixed.

File: uqcsbot/gaming.py (hunk 2 above)
Comment (bradleysigma) [reply uqcsbot/gaming.py:191]: Fixed.

File: uqcsbot/gaming.py (hunk 3 above)
Comment (bradleysigma) [reply uqcsbot/gaming.py]: Fixed.

File: uqcsbot/gaming.py (hunk 4 above)
Comment (bradleysigma) [reply uqcsbot/gaming.py:212]: Fixed.

File: uqcsbot/gaming.py (hunk 5 above)
Comment (bradleysigma) [reply uqcsbot/gaming.py]: Fixed.

**Review by JamesDearlove — APPROVED** [review]

**Pull request merged**